inference/core/utils/image_utils.py

```diff
@@ -87,35 +133,48 @@ def load_image_inferred(value: Any, cv_imread_flags=cv2.IMREAD_COLOR) -> Image.I
         NotImplementedError: If the image type could not be inferred.
     """
     if isinstance(value, (np.ndarray, np.generic)):
+        validate_numpy_image(data=value)
         return value, True
     elif isinstance(value, Image.Image):
         return np.asarray(value.convert("RGB")), False
     elif isinstance(value, str) and (value.startswith("http")):
-        return load_image_url(value, cv_imread_flags=cv_imread_flags), True
-    elif isinstance(value, str) and os.path.exists(value):
+        return load_image_from_url(value=value, cv_imread_flags=cv_imread_flags), True
+    elif isinstance(value, str) and os.path.isfile(value):
         return cv2.imread(value, cv_imread_flags), True
-    elif isinstance(value, str):
-        try:
-            return load_image_base64(value, cv_imread_flags=cv_imread_flags), True
-        except Exception:
-            pass
-        try:
-            return load_image_multipart(value, cv_imread_flags=cv_imread_flags), True
-        except Exception:
-            pass
-        try:
-            return load_image_numpy_str(value), True
-        except Exception:
-            pass
-    raise NotImplementedError(
-        f"Could not infer image type from value of type {type(value)}."
-    )
-
-
-pattern = re.compile(r"^data:image\/[a-z]+;base64,")
+    else:
+        return attempt_loading_image_from_string(
+            value=value, cv_imread_flags=cv_imread_flags
+        )
 
 
-def load_image_base64(value: str, cv_imread_flags=cv2.IMREAD_COLOR) -> np.ndarray:
+def attempt_loading_image_from_string(
+    value: Union[str, bytes, bytearray, _IOBase],
+    cv_imread_flags: int = cv2.IMREAD_COLOR,
+) -> Tuple[np.ndarray, bool]:
+    try:
+        return load_image_base64(value=value, cv_imread_flags=cv_imread_flags), True
+    except:
+        pass
+    try:
+        return (
+            load_image_from_encoded_bytes(value=value, cv_imread_flags=cv_imread_flags),
+            True,
+        )
+    except:
+        pass
+    try:
+        return (
+            load_image_from_buffer(value=value, cv_imread_flags=cv_imread_flags),
+            True,
+        )
+    except:
+        pass
+    return load_image_from_numpy_str(value=value), True
```

Comment:
Should we do one last try/except and raise an exception saying we are unable to load image from string?

Reply:
we could, but afaik `load_image_from_numpy_str()` raises `InvalidNumpyInput` in case of failure, which is then handled by HTTP handler (`InvalidNumpyInput` is dependent class of `InputImageLoadError`) and gives out HTTP 400

Reply:
ok, no problem